packages/gasket-plugin-typescript/EXAMPLES.md

```diff
@@ -0,0 +1,260 @@
+
+## Create Hook Usage
+
+The plugin's `create` hook automatically runs during app creation. Here are examples of how it affects different app types:
+
+### API App Configuration
+
+```js
+// Results in these package.json additions for API apps:
+{
+  "devDependencies": {
+    "tsx": "^4.19.3",
+    "typescript": "^5.8.2"
+  },
+  "scripts": {
+    "prebuild": "tsx gasket.ts build",
+    "build": "tsc",
+    "preview": "npm run build && npm run start",
+    "start": "node dist/server.js",
+    "local": "tsx watch server.ts"
+  },
+  "eslintIgnore": ["dist"]
+}
+```
+
+### Next.js App Configuration
+
+```js
+// Results in these package.json additions for Next.js apps:
+{
+  "dependencies": {
+    "tsx": "^4.19.3",
+    "typescript": "^5.8.2"
+  },
+  "devDependencies": {
+    "concurrently": "^9.1.2"
+  }
+}
+```
+
+### Custom Server Next.js Configuration
+
+```js
+// For apps with nextServerType: 'customServer'
+// Generates both next/tsconfig.json and shared/server.ts files
+// Adds 'dist' to .gitignore and eslintIgnore
+```
+
+## Generated Files
+
+### API App tsconfig.json
+
+```json
+{
+  "compilerOptions": {
+    "skipLibCheck": true,
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "target": "ESNext",
+    "isolatedModules": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "outDir": "dist",
+    "lib": ["esnext"],
+    "types": ["node"],
+    "baseUrl": "./",
+    "allowJs": true,
+    "strict": false,
+    "incremental": true,
+    "resolveJsonModule": true
+  },
+  "exclude": ["node_modules"],
+  "include": [
+    "./plugins",
+    "./routes",
+    "gasket-data.ts",
+    "server.ts",
+    "gasket.ts"
+  ]
+}
+```
+
+### Next.js App tsconfig.json
+
+```json
+{
+  "compilerOptions": {
+    "skipLibCheck": true,
+    "module": "ESNext",
+    "moduleResolution": "bundler",
+    "target": "ESNext",
+    "isolatedModules": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "allowImportingTsExtensions": true,
+    "outDir": "dist",
+    "lib": ["esnext", "dom"],
+    "types": ["node"],
+    "baseUrl": ".",
+    "paths": {
+      "@/gasket": ["./dist/gasket.js"]
+    },
+    "allowJs": true,
+    "strict": false,
+    "noEmit": true,
+    "incremental": true,
+    "resolveJsonModule": true,
+    "jsx": "preserve",
+    "plugins": [{ "name": "next" }]
+  },
+  "exclude": ["node_modules"],
+  "include": [
+    "./routes",
+    "./plugins",
+    "**/*.ts",
+    "**/*.tsx",
+    ".next/types/**/*.ts"
+  ]
+}
+```
+
+### Next.js App tsconfig.server.json
+
+**When generated:** Only for Next.js apps with custom server or HTTPS proxy setup.
+
+**Why needed:** Separate configuration for server-side TypeScript compilation vs client-side bundling.
+
+**Conditions for inclusion:**
+- `nextServerType === 'customServer'` OR
+- `nextDevProxy === true`
+
+**NOT included when:**
+- Standard Next.js setup without custom server or dev proxy
+
+```json
+{
```

Comment:
Again, do we need the entire config? Or can it just highlight what might be different?